### PR TITLE
Models creation issue #2

### PR DESCRIPTION
**Pull request by ALAN1MACIAS**
Created the models with the required attributes and relations.
Models:
CompanyBranch: manage the company branches.
Employee: manage the basic employee information and add association to a company branch.
AccessPoint: manage the checkin-checkout of the employees.

# Description
This issue add the models and the relationships of them.
- An employee  is associate with a company Branch.
- An access point is associate with an employee to track the checkin-checkout process of this.

### CODE REVIEW

File: app/models/access_point.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Indexes
+#
+#  index_access_points_on_employee_id  (employee_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (employee_id => employees.id)
+#
+class AccessPoint < ApplicationRecord
+  enum status: { in: 0, out: 1 }
```
Comment (anxhe) [inline app/models/access_point.rb:21]: Interesting , I haven't contemplated this approach

File: app/models/employee.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#
+#  index_employees_on_company_branch_id  (company_branch_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (company_branch_id => company_branches.id)
+#
+class Employee < ApplicationRecord
+  belongs_to :company_branch
+  has_many :access_points
```
Comment (anxhe) [inline app/models/employee.rb:26]: 🧐

**Review by anxhe — APPROVED** [review]

**Comment by anxhe** [comment]
Nice job

**Comment by anxhe** [comment]
I see that you have created all models in this pull request , looks like a good idea to advance faster and can bifurcate your branches better, so you don't need to wait for reviews too long.

File: app/models/company_branch.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# == Schema Information
+#
+# Table name: company_branches
+#
+#  id         :bigint           not null, primary key
+#  address    :string
+#  name       :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
```
Comment (RobertVel666) [inline app/models/company_branch.rb:9]: Why do you have a created_at and a updated_at attribute? just curious since it isn't used in any reports

Comment (ALAN1MACIAS) [reply app/models/company_branch.rb:9]: In general, a table should always have these fields in order to have control over the information that is recorded and when it is updated.

Comment (RobertVel666) [reply app/models/company_branch.rb:9]: I see, thanks for the info

**Review by RobertVel666 — APPROVED** [review]
This design is great, you took in consideration the relations and the fact that there must be a record of Clock-ins and it completed the design of the DB, it looks pretty sleek. Great job.

**Pull request merged**